### PR TITLE
Add contract verification section

### PR DESCRIPTION
**Pull request by conor10**
I've created an initial draft for a contract verification section for the ink! docs.

I refer to this as a draft as whilst the technical details provided are accurate (although it does duplicate a title image), we want to ensure that the suggested approaches are appropriate for the ink! community. 

We reference the Sirato ink! contract explorer as it provides the most accessible UX for verifying contracts, but everything here is OSS and it would be useful to get the team's input on the verification approach that should be taken that best serves the needs of the ink! community.

For instance, we could remove references to Sirato and just use the verification web service, however, UX would suffer only by having this approach documented (the data ends up in the same place with both approaches).

Web3 Labs wants to ensure that there is a standardised approach to ink! contract verification for the DotSama community, much like [Sourcify](https://sourcify.dev/) in the Ethereum community, where a standardised solution is embraced throughout the ecosystem, instead of having multiple competing services resulting in fragmented contract verification data. 

The sooner we can get verification into the ink! developers' workflow the better, and as there is a working solution available, we want to encourage usage of it.

### CODE REVIEW

File: docs/basics/verification.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+<img src="/img/title/magnifying-glass.svg" className="titlePic" />
+
+# Contract Verification
+
+Contract verification is the process of matching a deployed ink! contract 
+with the source code and metadata generated when it was built. 
+
+The verification process for WASM-based smart contract languages is more 
+complex then EVM-based languages such as Solidity due to the Rust
```
Comment (SkymanOne) [inline docs/basics/verification.md:15]:
```suggestion
complex than EVM-based languages such as Solidity due to the Rust 
```

File: docs/basics/verification.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+# Contract Verification
+
+Contract verification is the process of matching a deployed ink! contract 
+with the source code and metadata generated when it was built. 
+
+The verification process for WASM-based smart contract languages is more 
+complex then EVM-based languages such as Solidity due to the Rust 
+compiler not providing deterministic builds of contracts. 
+
+In order to verify an ink! or any WASM-smart contract the verification
```
Comment (SkymanOne) [inline docs/basics/verification.md:18]:
```suggestion
In order to verify an ink! or any Wasm smart contract the verification 
```

File: docs/basics/verification.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+Contract verification is the process of matching a deployed ink! contract 
+with the source code and metadata generated when it was built. 
+
+The verification process for WASM-based smart contract languages is more 
+complex then EVM-based languages such as Solidity due to the Rust 
+compiler not providing deterministic builds of contracts. 
+
+In order to verify an ink! or any WASM-smart contract the verification 
+process must rebuild the contract in an identical environment to which it
```
Comment (SkymanOne) [inline docs/basics/verification.md:19]: Could you please rephrase this?

File: docs/basics/verification.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+Once deployed, you will need to make a note of the contract's code hash in 
+order to verify it.
+
+## Verifying an ink! smart contract
+
+### Using the verification service web app
+
+The [ink! Verification Service](https://github.com/web3labs/ink-verifier-server)
+is a RESTful web service created for verifying smart contracts deployed to WASM.
```
Comment (SkymanOne) [inline docs/basics/verification.md:140]: > deployed to WASM.

Do you mean _deployed on `pallet-contracts`_ or _deployed to Wasm_?

File: docs/basics/verification.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+The verification process entails the following steps:
+
+1. A requestor uploads the source packge archive for a network and code hash
+1. The server checks that:
+   - The source code for the network and code hash is not already verified or 
+     being verified
+   - There is enough host resources to start a new verification
+1. The server downloads the pristine WASM byte code correspondening to the 
+   provided network and code hash
+1. The server streams the archive if is a compressed archive
```
Comment (SkymanOne) [inline docs/basics/verification.md:158]:
```suggestion
1. The server streams the archive if it is a compressed archive
```

**Review by SkymanOne — CHANGES_REQUESTED** [review]
Very well-written and detailed document. I don't have any objections to mentioning Sirato as the code behind it is OS as you mentioned.

I found a few typos and unclear sentences that might need rephrasing.

**Comment by conor10** [comment]
Thanks for your feedback @SkymanOne, changes have been incorporated.

**Review by SkymanOne — APPROVED** [review]
Thanks! That looks way better.

**Pull request merged**